retail/retail-java-applications/terraform/test/README.md

```diff
@@ -0,0 +1,114 @@
+# Automated Test For Terraform Modules
+
+Use go and terratest to test the following modules.
+- Pub/Sub
+
+## Install Terraform and Go
+
+If you are using the [Google Cloud Shell](https://cloud.google.com/shell/docs/how-cloud-shell-works), both Terraform and Go are installed.
+
+Follow the instructions to install [Terraform cli](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/gcp-get-started) and [Go](https://golang.org/doc/install).
+
+This repo has been tested on Terraform version `0.14.5`, Go version `1.16` and Google provider version `3.48.0`
+
+## Terraform Folder Structure
+
+
+The terraform folder structure can be found below. 
+
+Note that the terraform folder is inside dataflow-sample-applications/retail/retail-java-applications folder.
+
+```
+terraform
+├── examples
+│   └── pubsub
+│       ├── main.tf
+│       ├── terraform.tfvars
+│       └── variables.tf
+├── modules
+│   └── pubsub
+│       ├── main.tf
+│       └── variables.tf
+├── test
+│   ├── go.mod
+│   ├── go.sum
+│   ├── pubsub_test.go
+│   └── README.md
+├── main.tf
+├── README.MD
+├── terraform.tfvars
+└── variables.tf
+```
+
+The examples folder contains terraform scripts that serves as living documentation about how to use the module. They are also used by automated tests for execution.
+
+The modules folder contains the scripts that form the basis of the module.
+
+The test folder contains automated test and required files for go module which allows the test to be run outside of GOPATH.
+
+The terraform folder contains terraform scripts which deploys required resources for retail java application using corresponding terraform modules.
```

Comment:
As an example, this should look like:
The `terraform` folder contains Terraform scripts which deploy the required resources for `retail-java-application`.